agent/csi/plugin.go

```diff
@@ -88,16 +140,23 @@ func (np *NodePlugin) GetPublishedPath(volumeID string) string {
 }
 
 func (np *NodePlugin) NodeGetInfo(ctx context.Context) (*api.NodeCSIInfo, error) {
+	np.mu.RLock()
+	defer np.mu.RUnlock()
 
-	resp := &csi.NodeGetInfoResponse{
-		NodeId: np.nodeID,
+	resp, err := np.Client().NodeGetInfo(ctx, &csi.NodeGetInfoRequest{})
+	if err != nil {
+		return nil, err
 	}
 
 	return makeNodeInfo(resp), nil
 }
 
 func (np *NodePlugin) NodeStageVolume(ctx context.Context, req *api.VolumeAssignment) error {
 
+	if !np.staging {
+		return status.Error(codes.Unimplemented, "Plugin doesn't support Stage")
+	}
+
 	volID := req.VolumeID
 	stagingTarget := fmt.Sprintf(TargetStagePath, volID)
```

Comment:
Unrelated to this PR, but it may be better to, instead of using `fmt.Sprintf`, use the `path/filepath` library.